lib/blockchains/AlgorandBlockchain.js

```diff
@@ -369,14 +365,15 @@ class AlgorandBlockchain {
     }
 
     if (withdrawalAddresses.coinAddress && result.tokensWithdrawalTx.status !== "failed") {
+      // wait timeout in order to not get too many requests error and let nodes to sync balance after previous tx
+      await new Promise(resolve => setTimeout(resolve, this.envs.algorandDisableWalletCoinBalanceReqTimeout * 1000))
+      const coinBalance = await this.getAlgoBalanceForHotWallet(hotWallet.address)
+
```

Comment:
I think we can avoid the issue by using `CloseRemainderTo` field, which enforces to send all of the remaining coins to specified address:

<img width="1158" alt="Screenshot 2023-01-27 at 18 54 33" src="https://user-images.githubusercontent.com/10757562/215146641-67913eee-aa59-417f-b614-91112ade09fc.png">


https://developer.algorand.org/docs/get-details/transactions/transactions/#payment-transaction

https://www.algorand.foundation/news/limited-key-rotation

Reply:
Thanks for idea. Inside `withdrawCoinsTx` we specify identical Receiver and CloseRemainderTo, so I think the amount must be less than available. I will try to set 0 amount and check